encode.go

```diff
@@ -61,24 +61,24 @@ func (e *encoder) init() {
 		e.indent = 4
 	}
 	e.emitter.best_indent = e.indent
-	yaml_stream_start_event_initialize(&e.event, yaml_UTF8_ENCODING)
+	yamlStreamStartEventInitialize(&e.event, yaml_UTF8_ENCODING)
```

Comment:
See https://github.com/yaml/go-yaml/pull/95#discussion_r2311929936

I feel like code should be

```suggestion
	e.event = newStreamStartEvent(yaml_UTF8_ENCODING)
```